test/suites/drift/suite_test.go

```diff
@@ -148,20 +150,84 @@ var _ = Describe("Drift", Label("AWS"), func() {
 			g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(node), &v1.Node{})).To(Succeed())
 		}).WithTimeout(time.Minute).Should(Succeed())
 	})
-})
+	It("should deprovision nodes that have drifted due to securitygroup", func() {
+		By("getting the cluster vpc id")
+		output, err := env.EKSAPI.DescribeCluster(&eks.DescribeClusterInput{Name: awssdk.String(settings.FromContext(env.Context).ClusterName)})
+		Expect(err).To(BeNil())
+
+		By("looking for security groups")
+		securitygroups := env.GetSecurityGroups(map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName})
+		testSecurityGroup, found := lo.Find(securitygroups, func(sg aws.SecurityGroup) bool {
+			return awssdk.StringValue(sg.GroupName) == "security-group-drift"
+		})
+
+		if !found {
```

Comment:
nit: This technically has a race where it could be created in-between. Not a big deal for a test scenario but I think it might be simpler to just attempt to create it and then ignore the AlreadyExists error if that's what comes back from the API